Source/WebCore/loader/cache/CachedImage.cpp

```diff
@@ -521,21 +516,10 @@ void CachedImage::updateBufferInternal(const FragmentedSharedBuffer& data)
 
     EncodedDataStatus encodedDataStatus = EncodedDataStatus::Unknown;
 
-    if (isPostScriptResource()) {
-#if PLATFORM(MAC) && !USE(WEBKIT_IMAGE_DECODERS)
-        // Delay updating the image with the PostScript data till all the data
-        // is received so it can be converted to PDF data.
-        return;
-#else
-        // Set the encodedDataStatus to Error so loading this image will be canceled.
-        encodedDataStatus = EncodedDataStatus::Error;
```

Comment:
You've changed the behavior a bit here, haven't you? Do we still want to cancel the load like we used to?

Reply:
(just, without an exception on macOS)

Reply:
I think this is fine. I do not think we need to have a special case for postscript images here. 

The call `updateImageData()` below will set the `Image` with the encoded data. If postscript is not supported, it will be treated as a `BitmapImage`. And in this case `ImageDecoder::encodedDataStatus()` will return `EncodedDataStatus::Error`.

Reply:
Excellent!